service/src/main/kotlin/org/opensearch/simpleschema/index/SimpleSearchIndex.kt

```diff
@@ -111,6 +111,40 @@ internal object SimpleSearchIndex {
         }
     }
 
+    private fun createIndexRequest(): CreateIndexRequest? {
+        val classLoader = SimpleSearchIndex::class.java.classLoader
+        val indexMappingSource = classLoader.getResource(SIMPLESCHEMA_MAPPING_FILE_NAME)?.readText()!!
+        val indexSettingsSource = classLoader.getResource(SIMPLESCHEMA_SETTINGS_FILE_NAME)?.readText()!!
+        val request = CreateIndexRequest(INDEX_NAME)
+            .mapping(indexMappingSource, XContentType.YAML)
+            .settings(indexSettingsSource, XContentType.YAML)
+        return request
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    private fun indexCreateCall() {
+        val request = createIndexRequest()
```

Comment:
How exactly does the current implementation avoid concurrency issues? If I understand right, we have every node eagerly creating a default index on cluster update, is there a lock somewhere that I'm missing?

Reply:
this has changes for using the LifecycleListener and the creation of an index is cluster wide...